tango_simlib/utilities/fandango_json_parser.py

```diff
@@ -58,6 +60,37 @@ def parse(self, json_file):
             elif data_component == "dev_class":
                 self.device_class_name = elements
 
+    def _flatten_device_attribute_dictionary(self):
+        for attribute_config in self._device_attributes.values():
+            _events_properties = attribute_config["events"]
+            events_properties = self._extract_attribute_events_properties(
+                _events_properties
+            )
+            attribute_config.update(events_properties)
+
+            alarms_properties = attribute_config["alarms"]
+            attribute_config.update(alarms_properties)
+
+    def _remove_unused_device_attributes_keys(self):
+        keys_to_pop = [
+            "alarms",
+            "color",
+            "database",
+            "device",
+            "dim_x",
+            "dim_y",
```

Comment:
For spectrum and images, do we determine them automatically when setting this initial value for the attribute, or do we just use the maximums?  I'm wondering what would happen if the size of the data in the `value` field didn't match the `max_dim_x` and `max_dim_y`.

Reply:
We only use the maximums. If the size of the data in the `value` field was to be bigger than the `max_dim_x` or `max_dim_y` we would get an error when we try to read the attribute:

```python
DevFailed[
DevError[
    desc = Data size for attribute achievedPointing exceeds given limit
  origin = Attribute::set_value()
  reason = API_AttrOptProp
severity = ERR]

DevError[
    desc = Failed to read_attribute on device mid_dish_0000/elt/master, attribute achievedPointing
  origin = DeviceProxy::read_attribute()
  reason = API_AttributeFailed
severity = ERR]
]
```
which occurs here https://github.com/ska-sa/tango-simlib/blob/250fa66201124db98f7b76ac5e73715eabc835da/tango_simlib/tango_sim_generator.py#L130-L141 in the library for SPECTRUM attributes.

Reply:
OK, thanks for the detail.